firmware/library/L2_Utilities/utility_test.cpp

```diff
@@ -0,0 +1,104 @@
+#include "L2_Utilities/utility.hpp"
+#include "L5_Testing/testing_frameworks.hpp"
+
+TEST_CASE("Testing Random Functions", "[random]")
+{
+    SECTION("SetRandomSeed Consistency")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = std::rand() % 100;
+        CHECK(randomtester1 == randomtester2);
+    }
+
+    SECTION("Random")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = utility::Random(0, 99);
+        CHECK(randomtester1 == randomtester2);
+    }
+}
+
+TEST_CASE("Testing Bit Manipulations", "[Bit Manipulation methods]")
+{
+    SECTION("BitExtract")
+    {
+        CHECK(utility::BitExtract(15, 2, 3) == 7);
+        CHECK(utility::BitExtract(15, 3, 3) == 7);
+        CHECK(utility::BitExtract(15, 31, 32) == 15);
+        CHECK(utility::BitExtract(15, 0, 1) == 1);
+        CHECK(utility::BitExtract(15, 31, 28) == 0);
```

Comment:
Rather than use the decimal version of these numbers, it would be more readable to write the values in binary. For example, `0b1111`, to illustrate that the value contains only 1s.

Reply:
It would also be great to include some tests with numbers that only include all 1s.